### PR TITLE
[client-go]: make test code close to it's testing target source

### PR DESCRIPTION
**Pull request by ideapark**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note none

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/client-go/kubernetes/timeout_test.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package kubernetes_test
```
Comment (MHBauer) [inline staging/src/k8s.io/client-go/kubernetes/timeout_test.go:17]: We could leave it in the _test pkg for isolation purposes while still moving the location of the files.

**Review by MHBauer — COMMENTED** [review]
Overall I don't think we need to do this.

File: staging/src/k8s.io/client-go/kubernetes/timeout_test.go (hunk 1 above)
Comment (iotty) [reply staging/src/k8s.io/client-go/kubernetes/timeout_test.go:17]: sorry, i dont quit understand you?

Comment (iotty) [reply staging/src/k8s.io/client-go/kubernetes/timeout_test.go:17]: For client-go was widely imported by other projects, people will frequently be
redirected to the client-go source package, and they will take a quick look
at the source folder. `kubernetes_test` would make noise for them to
deep further (i am one of the victim but not the last).

Comment (MHBauer) [reply staging/src/k8s.io/client-go/kubernetes/timeout_test.go:17]: Why would this be noise? It's a separate package.

from `go help test`, see also https://golang.org/cmd/go/#hdr-Test_packages
```
Test files that declare a package with the suffix "_test" will be compiled as a separate package, and then linked and run with the main test binary. 
```

Comment (MHBauer) [reply staging/src/k8s.io/client-go/kubernetes/timeout_test.go:17]: I may be confused

Comment (thinpark) [reply staging/src/k8s.io/client-go/kubernetes/timeout_test.go:17]: sorry for the confusion, the main purpose of this pr is to remove the **directory** kubernetes_test

**Pull request closed without merging**